src/hexEditorProvider.ts

```diff
@@ -364,10 +374,40 @@ export class HexEditorProvider implements vscode.CustomEditorProvider<HexDocumen
 						.map(s => document.readBuffer(s[0], s[1] - s[0])),
 				);
 				const flatParts = flattenBuffers(parts);
-				const encoded = message.asText
-					? new TextDecoder().decode(flatParts)
-					: base64.fromUint8Array(flatParts);
-				vscode.env.clipboard.writeText(encoded);
+				if (message.format !== undefined) {
+					switch (message.format) {
```

Comment:
Rather than a big switch, let's have an object mapping of formats to functions. It would look something like:

```ts
const copyAsFormats: { [K in CopyFormat]: (buffer: Uint8Array) => void } = {
  [CopyFormat.Hex]: copyAsHex,
  ...
```